server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityStatusAction.java

```diff
@@ -0,0 +1,127 @@
+
+public class ActivityStatusAction implements CeWsAction {
+  private final UserSession userSession;
+  private final DbClient dbClient;
+  private final ComponentFinder componentFinder;
+
+  public ActivityStatusAction(UserSession userSession, DbClient dbClient, ComponentFinder componentFinder) {
+    this.userSession = userSession;
+    this.dbClient = dbClient;
+    this.componentFinder = componentFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller
+      .createAction("activity_status")
+      .setDescription("Return CE activity related metrics.<br>" +
+        "Requires 'Administer System' permission or 'Administer' rights on the specified project.")
+      .setSince("5.5")
+      .setResponseExample(getClass().getResource("activity_status-example.json"))
+      .setInternal(true)
+      .setHandler(this);
+
+    action.createParam(PARAM_COMPONENT_ID)
+      .setDescription("Id of the component (project) to filter on")
+      .setExampleValue(Uuids.UUID_EXAMPLE_03);
+    action.createParam(PARAM_COMPONENT_KEY)
+      .setDescription("Key of the component (project) to filter on")
+      .setExampleValue(KeyExamples.KEY_PROJECT_EXAMPLE_001);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    ActivityStatusWsResponse activityStatusResponse = doHandle(toWsRequest(request));
+    writeProtobuf(activityStatusResponse, request, response);
+  }
+
+  private ActivityStatusWsResponse doHandle(ActivityStatusWsRequest request) {
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      Optional<ComponentDto> component = searchComponent(dbSession, request);
+      String componentUuid = component.isPresent() ? component.get().uuid() : null;
+      checkPermissions(componentUuid);
+      int pendingCount = dbClient.ceQueueDao().countByStatusAndComponentUuid(dbSession, CeQueueDto.Status.PENDING, componentUuid);
+      int failingCount = dbClient.ceActivityDao().countLastByStatusAndComponentUuid(dbSession, CeActivityDto.Status.FAILED, componentUuid);
+
+      return ActivityStatusWsResponse.newBuilder()
+        .setPending(pendingCount)
+        .setFailing(failingCount)
+        .build();
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private Optional<ComponentDto> searchComponent(DbSession dbSession, ActivityStatusWsRequest request) {
+    ComponentDto component = null;
+    if (hasComponentInRequest(request)) {
+      component = componentFinder.getByUuidOrKey(dbSession, request.getComponentId(), request.getComponentKey(), COMPONENT_ID_AND_KEY);
+    }
+    return Optional.fromNullable(component);
+  }
+
+  private void checkPermissions(@Nullable String componentUuid) {
+    if (componentUuid == null) {
+      userSession.checkPermission(GlobalPermissions.SYSTEM_ADMIN);
+    } else {
+      userSession.checkComponentUuidPermission(UserRole.ADMIN, componentUuid);
```

Comment:
as discussed, it should stay as is as `userSession.checkComponentUuidPermission(String,String)` does not accept a @Nullable component uuid and rightly so